File: 탐색 & 시뮬레이션/숫자만 추출.py
import sys
# sys.stdin = open("input.txt", "rt")

'''
def to_natural_num(string):
    result = ""
    for ch in string:
        if ch.isdigit():
            result += ch
    return int(result)


def get_divisor(n):
    cnt = 0
    for i in range(1, n + 1):
        if n % i == 0:
            cnt += 1
    return cnt

string = input()
natural_num = to_natural_num(string)
print(natural_num)
print(get_divisor(natural_num))
'''

string = input()

res = 0
for ch in string:
    if ch.isdecimal():
        res = res * 10 + int(ch)
cnt = 0
for i in range(1, res + 1):
    if res % i == 0:
        cnt += 1
print(res)
print(cnt)

File: 코드 구현력 기르기/숫자만 추출.cpp
#include <stdio.h>
using namespace std;

int main(void){
	// freopen("input.txt", "r", stdin);
	char str[51];
	int result = 0, cnt = 0, i;
	scanf("%s", str);
	
	for(i = 0; str[i] != '\0'; i++){
		if(str[i] >= 48 && str[i] <= 57){
			result = result * 10 + (str[i] - 48);
		}
	}
	printf("%d\n", result);
	
	for(i = 1; i <= result; i++){
		if(result % i == 0) cnt++;	
	}
	printf("%d", cnt);
	
	return 0;	
}
